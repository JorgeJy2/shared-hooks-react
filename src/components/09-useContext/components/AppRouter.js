import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom";
import { AboutScreen } from '../pages/AboutScreen';
import { HomeScreen } from '../pages/HomeScreen';
import { LoginScreen } from '../pages/LoginScreen';
import { Footer } from './Footer';
export const AppRouter = () => {
    return (
        <Router>
            <div>
                <Footer/>
                <Switch >
                    <Route exact path="/about" component={AboutScreen} />
                    <Route exact path="/login" component={LoginScreen} />
                    <Route exact path="/" component={HomeScreen} />
                    {/* <Route component={HomeScreen} /> */}
                    <Redirect to='/'/>
                </Switch>
            </div>
        </Router>
    )
}
