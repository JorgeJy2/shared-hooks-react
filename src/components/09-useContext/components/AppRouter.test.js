import { mount } from "enzyme";
import { UserContext } from "../context/UserContext";
import { AppRouter } from "./AppRouter";

describe('Test router', () => {
    const user = {
        name: 'jorge',
        email: 'jorge@gmail.com'
    }

    const wrapper = mount(

        <UserContext.Provider
            value={
                {
                    user
                }}>
            <AppRouter />
        </UserContext.Provider>);

    test('view', () => {
        expect(wrapper).toMatchSnapshot();
    });
});