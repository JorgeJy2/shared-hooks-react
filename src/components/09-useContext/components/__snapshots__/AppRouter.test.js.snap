// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test router view 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Reat-useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/login"
              onClick={[Function]}
            >
              login
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <h1>
      Home
    </h1>
    <hr />
    <pre>
      {
   "name": "jorge",
   "email": "jorge@gmail.com"
}
    </pre>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</div>
`;
